cellmlmanip/model.py

```diff
@@ -888,7 +888,7 @@ def _replace_derivatives(self, new_derivative):
             for argument in equation.rhs.atoms(sympy.Derivative):
                 if new_derivative['expression'] == argument:
                     # add new equation
-                    new_eqn = equation.subs(new_derivative['expression'], new_derivative['variable'])
```

Comment:
We're not using "wildcard-like matching, parsing of match, and conditional replacements" here, so why not use `xreplace`?

Reply:
xreplace takes a dictionary, and replace takes 2 arguments. As far as I understand it the no wildcard equivalent for `.subs(old, new`) is `.replace(old, new)` and the no-wildcard equivalent for `.subs({old1, new1, ..., oldn: newn})` is `.xreplace({old1, new1, ..., oldn: newn})` is

Reply:
Neither `subs` nor `xreplace` uses wildcards; only `replace` does that!

You can replace `subs(old, new)` with `xreplace({old: new})` if you don't need to worry about bound variables.

Where it might go wrong is if the 'old' was a state variable or time, since `xreplace` would change derivatives which we might not want. Though sometimes this might be intended!

Another case `xreplace` can't do is replacing e.g. `x*y` within `x*y*z`, because it only finds complete nodes in the expression tree that match, not partial nodes.

Here though we're replacing a derivative itself with a variable, so `xreplace` should be fine.

Reply:
Ok I see